File: backend/src/main/java/org/luxons/sevenwonders/actions/PrepareMoveAction.java
package org.luxons.sevenwonders.actions;

import javax.validation.constraints.NotNull;

import org.hildan.livedoc.core.annotations.types.ApiType;
import org.luxons.sevenwonders.doc.Documentation;
import org.luxons.sevenwonders.game.api.PlayerMove;

/**
 * The action to prepare the next move during a game.
 */
@ApiType(group = Documentation.GROUP_ACTIONS)
public class PrepareMoveAction {

    @NotNull
    private PlayerMove move;

    /**
     * @return the move to prepare
     */
    public PlayerMove getMove() {
        return move;
    }

    public void setMove(PlayerMove move) {
        this.move = move;
    }
}

File: sw-server/src/main/kotlin/org/luxons/sevenwonders/actions/PrepareMoveAction.kt
package org.luxons.sevenwonders.actions

import org.hildan.livedoc.core.annotations.types.ApiType
import org.hildan.livedoc.core.annotations.types.ApiTypeProperty
import org.luxons.sevenwonders.doc.Documentation
import org.luxons.sevenwonders.model.PlayerMove

/**
 * The action to prepare the next move during a game.
 */
@ApiType(group = Documentation.GROUP_ACTIONS)
class PrepareMoveAction(
    /**
     * The move to prepare.
     */
    @ApiTypeProperty(required = true)
    val move: PlayerMove
)
